core/src/main/java/com/linecorp/armeria/common/HttpHeadersBase.java

```diff
@@ -174,6 +184,32 @@ URI uri() {
         }
     }
 
+    @Nullable
+    Locale selectLocale(Collection<Locale> supportedLocales) {
+        requireNonNull(supportedLocales, "supportedLocales");
+        return acceptLanguages()
+                .stream()
+                .flatMap(it -> Locale.filter(ImmutableList.of(it), supportedLocales).stream())
+                .findFirst()
+                .orElse(null);
+    }
+
+    List<LanguageRange> acceptLanguages() {
+        final List<String> acceptHeaders = getAll(HttpHeaderNames.ACCEPT_LANGUAGE);
+        if (acceptHeaders.isEmpty()) {
+            // No 'Accept-Language' header means accepting everything.
+            return ACCEPT_LANGUAGE_WILDCARD;
+        }
+
+        final List<LanguageRange> acceptLanguages = new ArrayList<>(4);
+        for (String acceptHeader : acceptHeaders) {
+            acceptLanguages.addAll(LanguageRange.parse(acceptHeader));
```

Comment:
What should we do when we see a bad header? Raise an exception or silently ignore? Other methods, such as `contentType()`, ignore bad values. When all header values are invalid, should we treat it as `ACCEPT_LANGUAGE_WILDCARD`?

Reply:
Added test as well.